research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,213 @@
+| `thumb` | Enables replacing the component's `thumb` part with custom markup. | A default thumb implementation is provided. |
+
+#### Host Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### Slotted Content Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### CSS Parts
+
+*Example Table*
+
+| Part Name | Description |
+| --------- | ----------- |
+| `thumb` | Enables styling the `thumb` part without requiring it to be completely replaced via its slot. |
+
+---
+
+## Implementation Plan
+
+### States and Interactions
+
+*List key component states, valid state transitions, and how interactions trigger state transitions.*
+
+- *Are states changed as a result of keyboard? mouse? touch?*
+- *Are there corresponding keyboard interactions for mouse/touch interactions?*
+- *Are some events triggered on key down vs key press?*
+
+*Example Table*
+
+| State Group | States | Initial State | Description | Interaction/Transition |
+| ----------- | ------ | ------------- | ----------- | ---------------------- |
+| `disabled` | `true`/`false` | `false` | When `true`, disables the control, preventing user interaction and displaying the control with a disabled appearance. | No interaction. Controlled programmatically. |
+
+### Accessibility
+
+*Consider the accessibility of the component.*
+
+#### Keyboard Navigation and Focus
+
+- *Arrow vs tabbing key behavior*
+- *Modifier key effects (e.g. shift, ctrl, meta)*
+- *Does this component use focus delegation?*
+
+*Example Table*
+
+| Key | Description |
+| --- | ----------- |
+| Up/Right  | Increments the value of the component by calling the `increment` method. If the `shift` modifier is pressed, increases by 10x the `step` value. |
+| Down/Left | Decrements the value of the component by calling the `decrement` method. If the `shift` modifier is pressed, decreases by 10x the `step` value. |
+
+#### Form Input
+
+*Does this component integrate with form submission, form validation, etc.? Can integration be accomplish with standard APIs or are special work-arounds required?*
+
+#### Use with Assistive Technology
+
+*Are there any details about shadow dom, focus delegation, or aria attributes that need special attention?*
+
+### Globalization
+
+*Consider whether the component has any special globalization needs such as:*
+
+- *Special RTL handling*
+- *Swapping of internal icons/visuals*
+- *Localization*
+
+### Security
+
+*Are there any security implications surrounding the component?*
+
+### Performance
+
+*Are there any performance pitfalls or challenges with implementing the component? (examples below)*
+
+* If the component renders a loop of items, should certain areas be considered for virtualization?
+* If the component needs to measure an area before rendering, how will jank be avoided?
+* If any measuring is needed at all, can rAF techniques help queue measures and prevent synchronous reflows?
+
+### Dependencies
```

Comment:
See response above. Most of these sections are here to make the spec writer think through things thoroughly. Often times "features" or bugs are accidentally implemented because someone didn't think about the various implications of things, be it perf, security, globalization, etc.